doc/decisions/array.md

```diff
@@ -40,11 +38,30 @@ myarray/#5 = value5
   array = #5
 ```
 
+To lookup an array, first do `ksLookup(ks,"/myarray")` on the parent.
+With the last index you get from its metadata `array`, iterate over the children.
+A cascading lookup on every individual child is also needed to make sure that overrides on individual
+elements are respected.
+
+For example:
+
+```
+spec/myarray    # <- contains the specification for the array
+spec/myarray/#  # <- contains the specification for the array elements
+
+dir/myarray/#0  # <- not an array, just an override for user/myarray/#
+
+user/myarray    # <- with metadata array=#0, this would be the array we get
+user/myarray/#0
+
+system/myarray  # <- not found in cascading lookup, as user/myarray exists
+```
```

Comment:
This example is quite confusing and took me a while to understand. It might be better to first write down what is in the KDB and then explain step by step what is found by which lookup.

One problem I have with the whole thing is: `dir:/myarray/#0` is only an array as long as `user:/myarray` exists. If `user:/myarray` is deleted, `dir:/myarray/#0` suddenly isn't an array element any more.

IMO `dir:/myarray/#0` should only be an array element, if `dir:/myarray` has metadata `array = #0` (or bigger index). Alternatively, we could say arrays must always be specified in `spec:/`.

Reply:
> It might be better to first write down what is in the KDB and then explain step by step what is found by which lookup.

Yes, actually a whole tutorial on arrays is needed.

> IMO dir:/myarray/#0 should only be an array element, if dir:/myarray has metadata array = #0 (or bigger index).

Yes, I agree and you understood correctly.

Is this already implemented in the high-level API?